Fix the error caused by passing unused context in JiraSensor.poke

kianelbo opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
#### This tiny PR attempts to fix the error caused by https://github.com/apache/airflow/pull/23046
(original issue: #22810 )

kianelbo committed:
Fix the error caused by passing unused context in JiraSensor.poke

kianelbo committed:
Remove unused context argument in JiraSensor test

jhtimmins commented on airflow/providers/jira/sensors/jira.py:
```diff
@@ -60,7 +60,7 @@ def poke(self, context: 'Context') -> Any:
         jira_result = getattr(resource, self.method_name)(**self.method_params)
         if self.result_processor is None:
             return jira_result
-        return self.result_processor(context, jira_result)
```
Is there a reason we're not removing `context` from the `poke` method signature?

uranusjr replied:
`poke` on other sensors take `context`, so this probably should as well.

kianelbo committed:
Make `TestJiraSensor` more robust

The sample ticket object used in the test is mocked more realistically and the original field_checker_func is used.
